packages/boilerplate-generator-tests/test-lib.js

```diff
@@ -19,7 +20,8 @@ export async function generateHTMLForArch(arch, includeHead) {
       cacheable: true,
       url: '/packages/templating-runtime.js?hash=c18de19afda6e9f0db7faf3d4382a4c953cabe18&v="1"',
       size: 24132,
-      hash: 'c18de19afda6e9f0db7faf3d4382a4c953cabe18'
+      hash: 'c18de19afda6e9f0db7faf3d4382a4c953cabe18',
```

Comment:
I had been tempted (both today and [in the past](https://github.com/meteor/meteor/issues/8286#issuecomment-275488777)) to suggest that we switch to using a newer cipher for `hash` (which currently uses SHA-1, 40-bytes in length — in hex) and just use that for the SRI — relying on only a single hash! 🙌🤔

That temptation was originally based on me thinking we could go with SHA-256, which has a relatively short hexadecimal representation (64-bytes) and would still work fine as a filename (which `hash` is frequently used for in builds and when serving files to clients!).

Unfortunately, as hashes become more complex, they become longer and the hex representation of a SHA-512 hash would be 128-bytes — a super long filename.  (Maybe too long?)

Most problematically though, SRI requires `base64` encoding which, while shorter than 128-bytes for SHA-512, fails to be useful as a filename because of the base64 encoding (`/`, in particular), so my temptation seems to be well-prevented.

Reply:
I‘d recommend switching to SHA-512 and simply truncate it for the filenames. [This seems to be totally fine](https://crypto.stackexchange.com/questions/9435/is-truncating-a-sha512-hash-to-the-first-160-bits-as-secure-as-using-sha1). If more information needs to go into the filename, there’s Base58 or [UNMISTAKABLE_CHARS](https://github.com/meteor/meteor/blob/14c135668d197131fbaab36d962fcbee9f911056/packages/random/random.js#L88), that would give Base55.

Recoding the hash from hex to Base64 for SRI would be trivial.